src/version-parser.ts

```diff
@@ -0,0 +1,33 @@
+import fetch from "node-fetch";
+import {parseString} from "xml2js";
+
+export async function getVersionToUse(url:string, allowBeta:boolean):Promise<string> {
+    const response = await fetch(url);
+    const json = await xml2json(await response.text());
+
+    const releaseVersion:string = json.metadata.versioning[0].release[0];
+    if(allowBeta || !releaseVersion.includes("beta")){
+        return releaseVersion;
+    }
+
+    const allVersions:string[] = json.metadata.versioning[0].versions[0].version;
+    for (let i = allVersions.length - 1 ; i >= 0; i--) {
```

Comment:
this is probably easier to read:

```javascript
const lastNonBeta = allVersions.slice().reverse().find(version => !version.includes('beta'))
if (!lastNonBeta) {
  return ''
}
return lastNonBeta
```
note that `slice()` is needed if you don't want to mutate the original array